### ISSUE TITLE
non-ascii strings cause parameter help mechanism break

### ISSUE THREAD

**Issue by Yanpas** [microsoft/vscode-cpptools#1504]
* Operating System and version: ubuntu 16.04
    * VS Code : 1.19.3
    * C/C++ extension version: 0.14.6
code snippet:
```
using namespace std;
struct A {
	static A st(const char*, string x, int) {
		return {};
	}
	A(string, string) {

	}
	A() {

	}
	void foo() {}
};

int main(int , char**) {
	std::vector<A> a {
		A::st("aaaaaaaaaaaaaaa", "абвг")
	};
}
```
`абвгд` are cyrillic multibyte symbols.
Bug:
![peek 2018-01-31 16-35](https://user-images.githubusercontent.com/7345761/35625899-0ad7c36a-06a5-11e8-9afa-d944e35f1029.gif)

**Comment by sean-mcmanus** [comment]
This bug appears to be fixed with our pending 0.15.0 release. We're planning to release an insiders build today.

**Comment by sean-mcmanus** [comment]
Oops, the bug only repros with the "Default" IntelliSenseEngine and not the "Tag Parser", so the bug still exists.

**Comment by sean-mcmanus** [comment]
The bug repros with just (after the 3rd letter, not before).
`#include <string>
std::string s{ "абв" };
`.
It's a bug in our IntelliSense process, but the strange thing is VS 2017 doesn't repro this (the code is shared).

**Comment by sean-mcmanus** [comment]
We found a fix for the repro case provided and others simple scenarios, but it turns out the fix is more complicated and ended up breaking other IntelliSense situations so a fix won't be available for the February release (possibly, March, but we're not sure).

**Comment by sean-mcmanus** [comment]
Non-ascii text shouldn't break signatureHelp now with our latest Insiders release: https://github.com/Microsoft/vscode-cpptools/releases . If you're using VS Code Insiders or have http.proxySupport not set to "off", you might not get the 0.21.0-insiders2 update automatically with C_Cpp.updateChannel set to "Insiders".

**Comment by Yanpas** [comment]
It didn't help
![image](https://user-images.githubusercontent.com/7345761/50181529-b65d1800-031d-11e9-8e6d-3a09ab7ff683.png)

**Comment by sean-mcmanus** [comment]
@Yanpas You're hitting a different bug with VS Code's signatureHelp caused by missing parameter names -- it also repros when one parameter name is a subset of another -- I filed a bug on VS code about this a long time ago. The bug is unrelated to UTF-8 characters. Adding unique names to the params fixes the issue:

![image](https://user-images.githubusercontent.com/19859882/50183432-e9c08800-02c6-11e9-9d75-55802f7efa7d.png)

**Comment by sean-mcmanus** [comment]
See https://github.com/Microsoft/vscode/issues/21518 and https://github.com/Microsoft/vscode/issues/18782 , but it looks like my bug got duped and the dup'd TypeScript issue closed, but this issue still repros with C++, so I'll try filing another bug with VS Code.

UPDATE: I filed bug https://github.com/Microsoft/vscode/issues/65327 .

**Comment by Yanpas** [comment]
Yeah, it works
![image](https://user-images.githubusercontent.com/7345761/50203599-1af69200-0373-11e9-96e7-e726dfeeecb5.png)

Interestingly, but it **breaks** if I name arguments like this :)
![image](https://user-images.githubusercontent.com/7345761/50203634-3792ca00-0373-11e9-9275-9062a7023e5b.png)

**Comment by sean-mcmanus** [comment]
@Yanpas Yeah, that's the "subset" issue I described at https://github.com/Microsoft/vscode/issues/21518 -- hopefully VS Code will reopen/fix the issue.

**Comment by Yanpas** [comment]
Opened new one: https://github.com/Microsoft/vscode/issues/65513

I think this can be closed

**Comment by sean-mcmanus** [comment]
@Yanpas The VS Code team informed me of an API they added -- I didn't realize this because I wasn't on the thread. I've opened an issue to track this fix: https://github.com/Microsoft/vscode-cpptools/issues/2952 .

UPDATE: It's been fixed -- maybe for the December update unless it gets pushed back to January.

**Comment by sean-mcmanus** [comment]
The non-ASCII issue is fixed, but the subset issue was delayed to our next Insider release due to the VS Code 1.30 dependency requirement.

**Comment by sean-mcmanus** [comment]
The signature help bug has been fixed with 0.22.0-insiders. You can get it via setting the C_Cpp.updateChannel setting to "Insiders", or downloading it manually from https://github.com/Microsoft/vscode-cpptools/releases .
